src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -322,17 +346,6 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 								  "foreign tables.")));
 	}
 
-	/* check that the relation does not contain any rows */
```

Comment:
You're right, I'll bring this back for non-hash tables and master_create_distributed_table.